includes/class-wc-payments-payment-request-button-handler.php

```diff
@@ -251,12 +251,13 @@ public function get_product_price( $product, ?bool $is_deposit = null, int $depo
 		}
 
 		if ( ! is_numeric( $base_price ) || ! is_numeric( $sign_up_fee ) ) {
+			$error_message =             sprintf(
+                // Translators: %d is the numeric ID of the product without a price.
+                __( 'Express checkout does not support products without prices! Please add a price to product #%d', 'woocommerce-payments' ),
+                (int) $product->get_id()
+            );
 			throw new Invalid_Price_Exception(
-				sprintf(
-					// Translators: %d is the numeric ID of the product without a price.
-					__( 'Express checkout does not support products without prices! Please add a price to product #%d', 'woocommerce-payments' ),
-					$product->get_id()
-				)
+				esc_html( $error_message )
```

Comment:
Two things added here: 
1. escaping the prepared error message
2. explicitly cast product ID to `int` type (this one is IMO not necessarily but can be a good additional measure)